### PR TITLE
Make the SIOFrameWriter call to finish non-mandatory

### PR DESCRIPTION
**Pull request by tmadlener**
BEGINRELEASENOTES
- Make calling `finish` for the `SIOFrameWriter` non-mandatory. See #442 for other related changes.

ENDRELEASENOTES

### CODE REVIEW

**Review by jmcarcell — APPROVED** [review]

**Comment by tmadlener** [comment]
Could we wait for CI to finish before merging, regardless of how trivial the PR looks, please? This one was missing https://github.com/AIDASoft/podio/commit/654ad98b9997aa6c5a7598ea5e661c19f58a0461 (I pushed that to master directly now)

**Pull request merged**
